packages/api-gateway/src/routers/apiGatewayRouter.ts

```diff
@@ -78,7 +82,25 @@ const apiGatewayRouter = (
     .get(
       "/agreements/:agreementId/attributes",
       authorizationMiddleware([M2M_ROLE]),
-      async (_req, res) => res.status(501).send()
+      async (req, res) => {
+        const ctx = fromApiGatewayAppContext(req.ctx, req.headers);
+
+        try {
+          const attributes = await agreementService.getAgreementAttributes(
+            ctx,
+            req.params.agreementId
+          );
+
+          return res.status(200).json(attributes).send();
+        } catch (error) {
+          const errorRes = makeApiProblem(
+            error,
+            emptyErrorMapper, // TODO map errors
```

Comment:
Waiting to solve doubt about draft agreement - if we want to not return info when agreement is draft, I need to correctly map the corresponding error here...
Otherwise it's ok